sdk/core/Azure.Core/src/DynamicData/DynamicData.AllowList.cs

```diff
@@ -195,6 +197,12 @@ private static bool IsAnonymousType(Type type)
             {
                 return type.Name.StartsWith("<>f__AnonymousType");
             }
+
+            // Determines if type inherits from IModelSerializable<T> for any T
+            private static bool DoesImplementIModelSerializable(Type type)
+            {
+                return type.GetInterfaces().Any(i => i.GetGenericTypeDefinition() == typeof(IModelSerializable<>));
```

Comment:
Let's remove LINQ here and use a manual loop, please.  We generally try to avoid the overhead for LINQ constructs in all libraries, but especially Core.